dss-api.yml

```diff
@@ -63,17 +63,60 @@ paths:
+
+        # Index design
+
+        The metadata seach index is implemented as a [document-oriented database](https://en.wikipedia.org/wiki/Document-oriented_database) using [Elasticsearch](https://www.elastic.co/). The index stores all information relevant to a bundle within each bundle document, largely eliminating the need for [object-relational mapping](https://en.wikipedia.org/wiki/Object-relational_mapping). This design is optimized for queries that filter the data.
+
+        To illustrate this concept, say our index stored information on three entities, `foo`, `bar`, and `baz`. A foo can have many bars and bars can have many bazes. If we were to index bazes in a document-oriented design, the information on the foo a bar comes from and the bazes it contains are combined into a single document. A example sketch of this is shown below in [JSON-schema](https://en.wikipedia.org/wiki/JSON#JSON_Schema).
+
+        ```
+        {
+          "definitions": {
+            "bar": {
+              "type": "object",
+              "properties": {
+                "uuid": {
+                  "type": "string",
+                  "format": "uuid"
+                },
+                "foo": {
+                  "type": "object",
+                  "properties": {
+                    "uuid": {
+                      "type": "string",
+                      "format": "uuid"
+                    },
+                    ...
+                  }
+                },
+                "bazes": {
+                  "type": "array",
+                  "items": {
+                    "type": "string",
+                    "format": "uuid"
+                  }
+                },
+                ...
+              }
+            }
+          }
+        }
+        ```
+
+        This closely resembles the structure of DSS bundle documents: projects have many bundles and bundles have many files. Each bundle document is a concatenation of the metadata on the project it belongs to and the files it contains.
+
+        # Limitations to index design
+
+        There are limitations to the design of DSS's metadata search index. A few important ones are listed below.
+
+        * [Joins](https://en.wikipedia.org/wiki/Join_(SQL)) between bundle metadata must be conducted client-side
+        * Querying is schema-specific; fields or values changed between schema version will break queries that use those fields and values
+        * A new search index must be built for each schema version
+        * A lot of metadata is duplicated between documents
```

Comment:
Can you elaborate why that is a limitation? IMO, denormalization is an database design technique, and not necessarily bad or limiting.